### PR TITLE
arduinoOTA: init at 1.4.1

### PR DESCRIPTION
**Pull request by poelzi**
###### Description of changes

arduinoOTA is a official arduino tool to upload remote boards

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 232055` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>arduinoOTA</li>
  </ul>
</details>

**Review by Janik-Haag — COMMENTED** [review]
The application is missing the version string you might have todo something like [this](https://github.com/Janik-Haag/nixpkgs/blob/b475406db789413f7063b66c29ca31f3e8e74f4a/pkgs/servers/osmocom/libasn1c/default.nix#L21)
```
[nix-shell:~/.cache/nixpkgs-review/pr-232055]$ arduinoOTA -v

[nix-shell:~/.cache/nixpkgs-review/pr-232055]$ arduinoOTA --version
arduinoOTA Version: 0.0.0-git Commit:  Date:
```

**Comment by pbek** [comment]
Result of `nixpkgs-review pr 232055` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>arduinoOTA</li>
  </ul>
</details>

**Comment by poelzi** [comment]
@Janik-Haag fixed

File: pkgs/development/embedded/arduino/arduinoOTA/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    owner = "arduino";
+    repo = pname;
+    rev = "${version}";
+    hash = "sha256-HaNMkeV/PDEotYp8+rUKFaBxGbZO8qA99Yp2sa6glz8=";
+  };
+
+  vendorHash = null;
+
+  postPatch = ''
+    sed -i 's/\(versionString        = "\)/\1${version}/' version/version.go
```
Comment (Janik-Haag) [inline pkgs/development/embedded/arduino/arduinoOTA/default.nix:20]: In nixpkgs there is the [`substituteInPlace`](https://nixos.org/manual/nixpkgs/stable/#fun-substituteInPlace) utility (for examples grep through nixpkgs) which is preferred over sed.

File: pkgs/development/embedded/arduino/arduinoOTA/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    homepage = "https://github.com/arduino/arduinoOTA";
+    description = "arduinoOTA is a tool for uploading programs to Arduino boards over a network.";
```
Comment (Janik-Haag) [inline pkgs/development/embedded/arduino/arduinoOTA/default.nix:27]:
```suggestion
    description = "A tool for uploading programs to Arduino boards over a network";
```
missed this the first time, in nixpkgs descriptions the name of the program shouldn't be repeated and sentences are supposed to end without punctuation.

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
Just two small nitpicks, the other stuff looks good to me.

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Comment by Janik-Haag** [comment]
Actually I will merge this since no one else complaint in over a month

**Pull request merged**